spec/validations/manifest_spec.rb

```diff
@@ -864,4 +864,152 @@ def create_package(parameter_hash)
+              'name' => 'api_token',
+              'type' => 'text',
+              'secure' => false,
+              'scopes' => ['header']
+            }
+          ]
+        }
+        package = create_package(@manifest_hash)
+        errors = ZendeskAppsSupport::Validations::Manifest.call(package, validate_scopes_for_secure_parameter: false)
+        expect(errors.find { |e| e.key == :scopes_require_secure_parameter }).to be_nil
+      end
+
+      it 'should have an invalid_attributes error when scope attribute is present' do
+        @manifest_hash = {
+          'parameters' => [
+            {
+              'name' => 'api_token',
+              'type' => 'text',
+              'secure' => true,
+              'scopes' => ['header']
+            }
+          ]
+        }
+        package = create_package(@manifest_hash)
+        errors = ZendeskAppsSupport::Validations::Manifest.call(package, validate_scopes_for_secure_parameter: false)
+        expect(errors.map(&:to_s)).to include(/parameters\.\[name=api_token\] contains invalid attributes: \["scopes"\]/)
+      end
+    end
+
+    context 'when validate_scopes_for_secure_parameter is true' do
+      context 'when a parameter has scopes but is not secure' do
+        it 'should have an error requiring secure parameter for scopes' do
+          @manifest_hash = {
+            'parameters' => [
+              {
+                'name' => 'api_token',
+                'type' => 'text',
+                'secure' => false,
+                'scopes' => ['header']
+              }
+            ]
+          }
+          package = create_package(@manifest_hash)
+          errors = ZendeskAppsSupport::Validations::Manifest.call(package, validate_scopes_for_secure_parameter: true)
+          expect(errors.find { |e| e.key == :scopes_require_secure_parameter }).not_to be_nil
+        end
+      end
+    end
+
+    context 'when a parameter has scopes but is not secure' do
+      it 'should have an error requiring secure parameter for scopes' do
+        @manifest_hash = {
+          'parameters' => [
+            {
+              'name' => 'api_token',
+              'type' => 'text',
+              'secure' => false,
+              'scopes' => ['header']
+            }
+          ]
+        }
+        expect(create_package(@manifest_hash)).to have_error(:scopes_require_secure_parameter)
+      end
+    end
+
+    context 'when a parameter has scopes and is secure' do
+      it 'should not have an error for valid scope values' do
+        @manifest_hash = {
+          'parameters' => [
+            {
+              'name' => 'api_token',
+              'type' => 'text',
+              'secure' => true,
+              'scopes' => ['header', 'body']
+            }
+          ]
+        }
+        expect(create_package(@manifest_hash)).not_to have_error(:scopes_require_secure_parameter)
+        expect(create_package(@manifest_hash)).not_to have_error(:invalid_secure_parameter_scopes)
```

Comment:
Should this test assert an empty errors array like the `should not have any errors` example?